### ISSUE TITLE
"AssertionError" & "AttributeError: 'NoneType' object has no attribute 'dtype'"

### ISSUE THREAD

**Issue by RECKENEFIN** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#763]
```
Traceback (most recent call last):
  File "D:\RVC\infer-web.py", line 201, in vc_single
    audio_opt = vc.pipeline(
  File "D:\RVC\vc_infer_pipeline.py", line 349, in pipeline
    self.vc(
  File "D:\RVC\vc_infer_pipeline.py", line 201, in vc
    score, ix = index.search(npy, k=8)
  File "D:\RVC\runtime\lib\site-packages\faiss\__init__.py", line 341, in replacement_search
    assert d == self.d
AssertionError

Traceback (most recent call last):
  File "D:\RVC\runtime\lib\site-packages\gradio\routes.py", line 321, in run_predict
    output = await app.blocks.process_api(
  File "D:\RVC\runtime\lib\site-packages\gradio\blocks.py", line 1007, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "D:\RVC\runtime\lib\site-packages\gradio\blocks.py", line 953, in postprocess_data
    prediction_value = block.postprocess(prediction_value)
  File "D:\RVC\runtime\lib\site-packages\gradio\components.py", line 2076, in postprocess
    processing_utils.audio_to_file(sample_rate, data, file.name)
  File "D:\RVC\runtime\lib\site-packages\gradio\processing_utils.py", line 206, in audio_to_file
    data = convert_to_16_bit_wav(data)
  File "D:\RVC\runtime\lib\site-packages\gradio\processing_utils.py", line 219, in convert_to_16_bit_wav
    if data.dtype in [np.float64, np.float32, np.float16]:
AttributeError: 'NoneType' object has no attribute 'dtype'
```

I've looked through and seen similar issues being submitted. I got this error while trying to run a conversion. Happens on both a fresh install on my local pc and a fresh download of the v2 notebook in Google Colab. Strangely, I got it to run just fine earlier in the day.

I tried multiple of the proposed fixes in the other threads:
- Check the audio file path/make sure it includes the file extension
- Set the "Voiceless/Breath" protector to 0.5
- Used a different model that I knew worked in the past
- etc.

and none of them worked.

So I thought to myself, "What did I do differently earlier today?" And the only difference was that, earlier, I ran my first conversion without doing any transposition and then adjusted it afterwards. This time, though, knowing what the transposition was, I skipped straight to the adjustment, and that's when the errors started getting thrown.

So I set up everything the way I had been trying (same models, audio, etc.) but kept the transpose set to 0. And, lo and behold, no errors and a successful conversion. Ran it again with the transpose set to 4. Successful conversion.

Just to verify what I was seeing, I closed the whole thing down and reopened go-web.bat. Same setup, set my first conversion with a transpose of 4. Errors.

The linked Colab located [here](https://colab.research.google.com/github/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/main/Retrieval_based_Voice_Conversion_WebUI.ipynb) never had these errors, so my guess is something got messed up in newer versions.

**TL;DR: If you're getting these errors, try running your initial conversion with transpose set to "0" and then adjust it afterwards.**

**Comment by RVC-Boss** [comment]
it's caused by you train a v1/v2 model, and the index is v2/v1 version

**Comment by RECKENEFIN** [comment]
> it's caused by you train a v1/v2 model, and the index is v2/v1 version

Hmm, that might be the case for the one model I used, but the other model I tried exhibited the same errors, and that was definitely a v1/v1 model/index.

I'll have to look into that all the same.

**Comment by RECKENEFIN** [comment]
Retrained the problem model just to make sure the versions matched up, and that has indeed solved the problem. Thank you for pointing that out.

**Comment by GH0STSecurity** [comment]
guys i still couldnt solve it, someone help me please :((
```
Traceback (most recent call last):
  File "G:\Artificial_Intelilgence\Mangio-RVC-v23.7.0\runtime\lib\site-packages\gradio\routes.py", line 437, in run_predict
    output = await app.get_blocks().process_api(
  File "G:\Artificial_Intelilgence\Mangio-RVC-v23.7.0\runtime\lib\site-packages\gradio\blocks.py", line 1349, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "G:\Artificial_Intelilgence\Mangio-RVC-v23.7.0\runtime\lib\site-packages\gradio\blocks.py", line 1283, in postprocess_data
    prediction_value = block.postprocess(prediction_value)
  File "G:\Artificial_Intelilgence\Mangio-RVC-v23.7.0\runtime\lib\site-packages\gradio\components.py", line 2587, in postprocess
    file_path = self.audio_to_temp_file(
  File "G:\Artificial_Intelilgence\Mangio-RVC-v23.7.0\runtime\lib\site-packages\gradio\components.py", line 364, in audio_to_temp_file
    processing_utils.audio_to_file(sample_rate, data, filename, format=format)
  File "G:\Artificial_Intelilgence\Mangio-RVC-v23.7.0\runtime\lib\site-packages\gradio\processing_utils.py", line 156, in audio_to_file
    data = convert_to_16_bit_wav(data)
  File "G:\Artificial_Intelilgence\Mangio-RVC-v23.7.0\runtime\lib\site-packages\gradio\processing_utils.py", line 170, in convert_to_16_bit_wav
    if data.dtype in [np.float64, np.float32, np.float16]:
AttributeError: 'NoneType' object has no attribute 'dtype'
```

**Comment by fiore202010** [comment]
Here is the solution of the "model inference" issue (unable to find/select inverencing voice): 

https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/pull/889/commits/354db27c1b66348913231d663061aeb04b255ceb

**Comment by grayleeeeee** [comment]
> 这是因为你训练了一个 v1/v2 模型，并且索引是 v2/v1 版本

当我更换算法到harvest，它是正常工作的
